rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.template

```diff
@@ -380,7 +380,9 @@ static message_type_support_callbacks_t callbacks = {
   "@(spec.base_type.type)",
   &register_type__@(spec.base_type.type),
   &publish__@(spec.base_type.type),
-  &take__@(spec.base_type.type)
+  &take__@(spec.base_type.type),
+  nullptr,  // convert ros to dds (handled differently for C++)
```

Comment:
The other problem with specializing this structure is that the `rmw_opensplice_cpp` stuff currently doesn't know or care if the incoming typesupport is C or C++, but if we had different data for each it would need to know so that it could cast it to the correct type. I think that's probably the main reason I avoided changing this originally.